invoice/src/main/java/org/killbill/billing/invoice/usage/ContiguousIntervalConsumableUsageInArrear.java

```diff
@@ -242,20 +246,19 @@ UsageConsumableInArrearTierUnitAggregate computeToBeBilledConsumableInArrearWith
         for (final TieredBlock tieredBlock : tieredBlocks) {
 
             tierNum++;
-            final int blockTierSize = tieredBlock.getSize().intValue();
-            final int tmp = remainingUnits / blockTierSize + (remainingUnits % blockTierSize == 0 ? 0 : 1);
+            final long blockTierSize = tieredBlock.getSize().longValue();
+            final long tmp = remainingUnits / blockTierSize + (remainingUnits % blockTierSize == 0 ? 0 : 1);
             if ( tmp > tieredBlock.getMax()) { /* Includes the case where max is unlimited (-1) */
-                remainingUnits -= tieredBlock.getMax() * blockTierSize;
+                remainingUnits -= tieredBlock.getMax().longValue() * blockTierSize;
             } else {
                 targetBlock = tieredBlock;
                 targetTierNum = tierNum;
                 break;
             }
         }
         final int lastBlockTierSize = targetBlock.getSize().intValue();
-        final int nbBlocks = units.intValue() / lastBlockTierSize + (units.intValue() % lastBlockTierSize == 0 ? 0 : 1);
-
-        return new UsageConsumableInArrearTierUnitAggregate(targetTierNum, targetBlock.getUnit().getName(), targetBlock.getPrice().getPrice(getCurrency()), targetBlock.getSize().intValue(), nbBlocks);
+        final long nbBlocks = units.intValue() / lastBlockTierSize + (units.intValue() % lastBlockTierSize == 0 ? 0 : 1);
```

Comment:
Do we want `units.intValue();` here?

Reply:
Nope, you right, fixed in https://github.com/killbill/killbill/pull/1328/commits/c3548aede500713383b62773b5c7a9e4d7e2f1a4